hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AbstractAsyncSocket.java

```diff
@@ -137,21 +137,17 @@ public final void close(String reason, Throwable cause) {
         this.closeReason = reason;
         this.closeCause = cause;
 
-        if (cause == null) {
```

Comment:
The original code structure I find a lot easier to read. If you agree with this, then such a refactoring makes the code worse, not better.

Reply:
I prefer the original one.

```
 if (cause == null) {
            if (logger.isInfoEnabled()) {
                if (reason == null) {
                    logger.info("Closing  " + this);
                } else {
                    logger.info("Closing " + this + " due to " + reason);
                }
            }
        } else {
            if (logger.isWarningEnabled()) {
                if (reason == null) {
                    logger.warning("Closing  " + this, cause);
                } else {
                    logger.warning("Closing " + this + " due to " + reason, cause);
                }
            }
        }
        ```
        Vs
        
        ```
        if (cause == null && logger.isInfoEnabled()) {
            if (reason == null) {
                logger.info("Closing  " + this);
            } else {
                logger.info("Closing " + this + " due to " + reason);
            }
        } else if (logger.isWarningEnabled()) {
            if (reason == null) {
                logger.warning("Closing  " + this, cause);
            } else {
                logger.warning("Closing " + this + " due to " + reason, cause);
            }
        }
        ```
        
        Perhaps I'm an old fart?

Reply:
Reverted https://github.com/hazelcast/hazelcast/pull/23740/commits/7483a2aca1f78971e2c5efd0d99d1d8b2dcbd201